packages/react-components/react-tree/src/components/Tree/Tree.types.ts

```diff
@@ -9,6 +9,7 @@ export type TreeSlots = {
 
 // eslint-disable-next-line @typescript-eslint/naming-convention
 export type TreeNavigationData_unstable =
+  | { event: React.MouseEvent<HTMLElement>; target: HTMLElement; type: 'Click' }
```

Comment:
Click as a navigation becomes required since for roving tab indexes on a click event the tabindex of the clicked element should be updated.